File: tests/examples/add_one_clip_in_each_layer.c
#include <ges/ges.h>

static void
add_one_clip_in_each_layer (GESTimeline * timeline, GstClockTime duration)
{
  GList *tmp, *layers = ges_timeline_get_layers (timeline);

  for (tmp = layers; tmp; tmp = tmp->next) {
    GESClip *clip = GES_CLIP (ges_test_clip_new ());
    GESLayer *layer = GES_LAYER (tmp->data);

    ges_layer_add_clip (layer, clip);
    g_object_set (clip, "duration", duration, "start", 0, NULL);
  }

  /* We commit the timeline to have our changes taken into account */
  ges_timeline_commit (timeline);
}

File: examples/python/add_one_clip_in_each_layer.py
from gi.repository import GES

def add_one_clip_in_each_layer (timeline, duration):
    for layer in timeline.get_layers():
        clip = GES.TestClip.new()
        layer.add_clip (clip)
        clip.props.duration = duration
        clip.props.start = 0

    # We commit the timeline to have our changes taken into account
    timeline.commit()
